### PR TITLE
[Relax][Op] More new relax operators!

### PR DESCRIPTION
**Pull request by SiriusNEO**
This PR adds:
- `relax.where` (transform)
- `relax.less` (tensor, binary)
- `relax.sigmoid` (tensor, unary)
- `relax.nn.cross_entropy` (nn)
- `relax.nn.softmax_cross_entropy` (nn)

Tests about their creation and legalization are all passed.

PS: `relax.where` is a ternary broadcast op. Given that currently there is no good solution to do runtime shape inference, I leave this case and return `RuntimeDepShape` for now. Further discussions and solutions to this issue is needed.

### CODE REVIEW

File: python/tvm/relax/op/tensor.py [hunk 1]
```diff
@@ -174,6 +174,40 @@ def sqrt(data: Expr) -> Expr:
...
+
+    Returns
+    -------
+    result : relax.Expr
+        The computed result.
+    """
+    return _ffi_api.sigmoid(data)
+
+
+def less(lhs, rhs):
```
Comment (MasterJH5574) [inline python/tvm/relax/op/tensor.py]: Let’s add the type annotation here.

File: python/tvm/relax/op/transform.py [hunk 2]
```diff
@@ -511,3 +511,44 @@ def convert_int(arr):
     end = convert_int(end)
     strides = convert_int(strides) if strides else None
     return _ffi_api.strided_slice(data, begin, end, strides, axes, slice_mode)
+
+
+def where(condition, x, y):
```
Comment (MasterJH5574) [inline python/tvm/relax/op/transform.py]: Ditto

File: python/tvm/relax/transform/op_legalizer.py [hunk 3]
```diff
@@ -290,6 +302,20 @@ def _nn_adaptive_max_pool2d(bb: BlockBuilder, args: List[Expr], attrs: Attrs, ou
     )
 
 
+def cross_entropy(x, y):
```
Comment (MasterJH5574) [inline python/tvm/relax/transform/op_legalizer.py]: Let’s use name `_cross_entropy` since the prepending underscore means this function is used internally.

File: tests/python/relax/test_relax_tensor_ops.py [hunk 4]
```diff
@@ -605,5 +605,64 @@ def expected(x: R.Tensor((2, 64, 8, 9), "float32")) -> R.Tensor(None, "float32",
...
+        bb = relax.BlockBuilder()
+        with bb.function("main", [predictions, targets]):
+            gv = bb.emit(relax_op_nn_func(predictions, targets))
+            bb.emit_func_output(gv)
+
+        expected = expected.with_attr("global_symbol", "main")
+        tvm.ir.assert_structural_equal(bb.get()["main"], expected)
+
+    _test_single_op(R.cross_entropy, relax.op.nn.cross_entropy)
+    _test_single_op(R.softmax_cross_entropy, relax.op.nn.softmax_cross_entropy)
```
Comment (MasterJH5574) [inline tests/python/relax/test_relax_tensor_ops.py]: Would be better to split them separately, since cross entropy and softmax cross entropy are not exactly the same thing.

**Comment by MasterJH5574** [comment]
Let’s do a rebase given that https://github.com/mlc-ai/relax/pull/43 is just merged

File: python/tvm/relax/transform/op_legalizer.py (hunk 3 above)
Comment (SiriusNEO) [reply python/tvm/relax/transform/op_legalizer.py]: Oh, but I noticed that the similar functions (e.g. line: 126, `expand_dims`) are named without prepending underscore

Comment (SiriusNEO) [reply python/tvm/relax/transform/op_legalizer.py]: A difference maybe that this function `cross_entropy` is shared by `_nn_cross_entropy` and `_nn_softmax_cross_entropy`, so I define it outside

Comment (MasterJH5574) [reply python/tvm/relax/transform/op_legalizer.py]: Yes that is because the `expand_dims` is already an internal one.

Comment (SiriusNEO) [reply python/tvm/relax/transform/op_legalizer.py]: But `_cross_entropy` defined outside looks like a function that returns `bb.call_te` (like `_relu`, `_sum`), will it cause confusions? In this case `cross_entropy` has a prefix `nn`. But if the similar situation happens in a normal op, we need to use another naming rule.

Comment (SiriusNEO) [reply python/tvm/relax/transform/op_legalizer.py]: We use `_topi_cross_entropy` before. Is it a good idea?

Comment (MasterJH5574) [reply python/tvm/relax/transform/op_legalizer.py]: Yeah sounds good.

File: python/tvm/relax/transform/op_legalizer.py [hunk 5]
```diff
@@ -340,6 +352,20 @@ def _nn_adaptive_max_pool2d(bb: BlockBuilder, args: List[Expr], attrs: Attrs, ou
...
+    return -topi.sum(topi.log(x) * y)
+
+
+def _nn_cross_entropy(bb: BlockBuilder, args: List[Expr], attrs: Attrs, output_shape: Expr):
+    return bb.call_te(cross_entropy, args[0], args[1])
+
+
+def _nn_softmax_cross_entropy(bb: BlockBuilder, args: List[Expr], attrs: Attrs, output_shape: Expr):
+    def softmax_cross_entropy(x, y):
+        return cross_entropy(topi.nn.softmax(x), y)
```
Comment (MasterJH5574) [inline python/tvm/relax/transform/op_legalizer.py]: :eyes:
```suggestion
        return _cross_entropy(topi.nn.softmax(x), y)
```

Comment (SiriusNEO) [reply python/tvm/relax/transform/op_legalizer.py]: Ahh I start to rebase and forget to change this.
But now I use the `te_` prefix to name the internal functions (see the lastest commit). How about this? 👀

**Review by MasterJH5574 — APPROVED** [review]
LGTM. Thanks for upstreaming!

File: python/tvm/relax/transform/op_legalizer.py (hunk 5 above)
Comment (MasterJH5574) [reply python/tvm/relax/transform/op_legalizer.py]: Yeah thanks for the change!

**Comment by SiriusNEO** [comment]
Thanks for your careful reviewing!
